lib/pure/unittest.nim

```diff
@@ -97,6 +138,40 @@ when declared(stdout):
 when not defined(ECMAScript):
   import terminal
 
+when declared(stdout):
+  const paralleliseTests* = existsEnv("NIMTEST_PARALLEL") or defined(nimtestParallel)
+    ## Whether parallel test running was enabled (set at compile time).
+    ## This constant might be useful in custom output formatters.
+else:
+  const paralleliseTests* = false
+
+when paralleliseTests:
+  import threadpool, locks
+
+  # repeatedly calling sync() without waiting for results - on procs that don't
+  # return any - doesn't work properly (probably due to gSomeReady getting its
+  # counter increased back to the pre-call value) so we're stuck with these
+  # dummy flowvars
+  var flowVars {.threadvar.}: seq[FlowVarBase]
+  flowVars = @[]
```

Comment:
That is not how thread local storage works.

Reply:
Reminder for you @Araq to give us a mechanism to write:

```nim
doThisOnEachThreadStart:
  flowVars = @[]
```

Otherwise it's a massive PITA.

Reply:
> That is not how thread local storage works.

Turns out I don't even have to initialise that nil sequence. The compiler takes care of it in each child thread, when nested tests are run (the only scenario in which it's used outside the main thread).